### PR TITLE
Gutenberg Tracking: Track List View open and close (wpcom_block_editor_list_view_toggle)

### PR DESCRIPTION
**Pull request by david-szabo97**
#### Changes proposed in this Pull Request

* Track "List View" toggle in both Post/Page, and Site Editor. Event name: `wpcom_block_editor_list_view_toggle`. A property called `is_open` is also added to the event. The property indicates the new state of the list view.

#### Testing instructions

* Follow steps to enable tracking debugging at PCYsg-nrf-p2
* Open Post Editor, click on the List View button in the header. Make sure `wpcom_block_editor_list_view_toggle` is fired
* Open Site Editor, click on the List View button in the header. Make sure `wpcom_block_editor_list_view_toggle` is fired
* Make sure `is_open` is set to `true` when you open the list view, and it's `false` when you close the list view.

Related to https://github.com/Automattic/wp-calypso/issues/53410

### CODE REVIEW

**Review by getdave — COMMENTED** [review]
I would say

1. Can we defintely not track via Redux actions?
2. You should add an e2e test to cover this as this DOM based tracking is very brittle.

See PCYsg-nrf-p2.

**Review by Addison-Stavlo — COMMENTED** [review]
Right now it seems this event `wpcom_block_editor_list_view_click` fires both when the list view is opened and closed.  Given this, I think we might need to add an extra property to the event that will convey whether the event opened or closed the list view.

Also noting the name `*_click` seems slightly inaccurate since this can be triggered by means such as keyboard shortcuts.  I wonder if we should name it `*_toggle`.

**Comment by Addison-Stavlo** [comment]
@getdave 

> Can we defintely not track via Redux actions?

What is the problem with tracking via Redux actions?  These seem like they would be more stable than selectors, as well as cover cases that selectors do not such as keyboard shortcuts, etc.

**Comment by david-szabo97** [comment]
Makes sense! Renamed the event and added the event property.

File: apps/wpcom-block-editor/src/wpcom/features/tracking.js [hunk 1]
```diff
@@ -352,6 +363,12 @@ const REDUX_TRACKING = {
 	'core/notices': {
 		createErrorNotice: trackErrorNotices,
 	},
+	'core/edit-site': {
+		setIsListViewOpened: trackListViewToggle,
+	},
+	'core/edit-post': {
```
Comment (getdave) [inline apps/wpcom-block-editor/src/wpcom/features/tracking.js:412]: I wonder given that the new Widgets screen will likely land in WP 5.8, whether you should add `edit-widgets` to this list for tracking (obviously check it fails silently as it's not in Core yet)?

Screen Shot 2021-06-09 at 09 50 48

Comment (david-szabo97) [reply apps/wpcom-block-editor/src/wpcom/features/tracking.js:412]: `edit-widgets` is still using the old dropdown list view. Which isn't tracked in the redux store.

**Review by Addison-Stavlo — APPROVED** [review]
Works great! 😁

**Comment by david-szabo97** [comment]
Rebased this for the new features we have. I'll add E2E tests too.

**Comment by getdave** [comment]
> I'll add E2E tests too.

🎉

**Review by Copons — APPROVED** [review]

**Comment by Copons** [comment]
Preapproved.
Feel free to ship once the E2E are in. 🙂

**Comment by Addison-Stavlo** [comment]
Im wondering if we should follow up to include tracking for a user selecting blocks via the list view? 🤔

**Comment by ianstewart** [comment]
I was also wondering about tracking deletion of blocks via list view. (Also, deletion of blocks via list view is super-cool.)

**Comment by Addison-Stavlo** [comment]
> I was also wondering about tracking deletion of blocks via list view. (Also, deletion of blocks via list view is super-cool.)

I didn't realize this was a thing... <s>and cant currently figure out how to do so on my own. </s> 😅  Ohh the 'delete' button of course. lol, that is cool

**Comment by Addison-Stavlo** [comment]
> I was also wondering about tracking deletion of blocks via list view. (Also, deletion of blocks via list view is super-cool.)

@ianstewart  - Looking further, the list view doesn't actually enable deletion of blocks.  The 'del' shortcut is something that works to delete a selected block (given the blocks text-input is not active), regardless of the list-view.  The list view just enables us to select the block.

So more things we could track with the list-view itself seem limited to:
* Selecting a block via the list-view.
* Hovering an item in the list view to see its outline in the editor.  Although, this seems like it would just be event spam and too much noise to be worth tracking.

**Comment by ianstewart** [comment]
> Hovering an item in the list view to see its outline in the editor. Although, this seems like it would just be event spam and too much noise to be worth tracking.

Definitely seems like event spam. 

> Selecting a block via the list-view.

Selecting seems worthwhile to track for usage.

**Pull request merged**